### ISSUE TITLE
Pin to taskbar not working on Windows 10 Build 1809

### ISSUE THREAD

**Issue by ckck2000** [lordmulder/stdutils#23]
Hello,

The plugin's ShellVerb.PinToTaskbar function doesn't work on Windows 10 Build 1809 (17763.379)
There was no problem on Windows 10 build 1803.

**Comment by lordmulder** [comment]
Yeah, this seems to be a bug (or "intended behavior") in Windows v1809 🙄 

All I know is that finding the **`FolderItemVerb`** for _"Pin to taskbar"_ is found successfully, but then **`FolderItemVerb::DoIt()`** fails for no apparent reason 😠

**Comment by gbelmont22** [comment]
I was just made aware my NSIS installs are no longer correctly pinning the shortcut to the taskbar and suspect this is the issue.  I did some digging and found this article:

[Pinning a network program with a Jump List to the taskbar programmatically in windows 10 – Part 1](http://alexweinberger.com/main/pinning-network-program-taskbar-programmatically-windows-10/)

The author describes the issue in Windoze 10 v1809 and outlines a methodology for fooling windows back into behaving itself.  I think there is enough information presented that you might be able (if so inclined) to cobble together a work around in StdUtils using the same technique.

**Comment by lordmulder** [comment]
Well, according to that article, the "solution" seems to be to rename your program to `explorer.exe`, in order to fool Windows – which apparently seems to "block" all other processes now. I haven't tried that yet, but it probably is **not** really feasible for most installers to be called `explorer.exe`. Also, the other approach, which is to manipulate the image path name, seems pretty awkward to me...

**Comment by gbelmont22** [comment]
Yes, you're right.  - I just skimmed the code and didn't really think about what they were doing.  For some reason, I had it in my head that they were changing the name of the process in memory (not the executable on the filesystem) - changing something in memory temporarily seemed like a workable hack, but I agree changing the name of the installer mid-install is not a good idea.  _Hooray for Micro$oft..._

**Comment by Nutzzz** [comment]
Note this feature could be revived based on the method that MS Edge is currently using (meaning it should stick around for a while).  This from <<https://geelaw.blog/entries/msedge-pins/>>:
> 1. Initialise COM.
> 2. Use **ILCreateFromPathW** to get an absolute PIDL.
> 3. Use **CoCreateInstance** to activate **CLSID_TaskbandPin** with interface **IID_IPinnedList3**.
> 4. Use **Modify** method with the PIDL to pin/unpin the target.
> 5. Release the interface, the PIDL and uninitialse COM.
>
> You can get an example [here](https://geelaw.blog/entries/msedge-pins/assets/pin.cc) under MIT license. Once compiled, use **pin "C:\path\to\file.lnk"** to pin a Shortcut, and **pin "C:\path\to\file.lnk" u** to unpin it.

**Comment by lordmulder** [comment]
Thanks, will definitely have a look at that!

I think the crucial point is the availability of the, apparently undocumented, **IID_IPinnedList3** interface.

Is it only available in latest Windows 10 or also in older OS?
